### PR TITLE
Speed is inverted when dist is negative

### PR DESCRIPTION
**Pull request by mdtoom**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   |  #1896 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Gazebo |

---

## Description of contribution in a few bullet points
* Fixes that backup_action_node cannot enter a negative speed and therefore the action cannot backed up.
* Changed to invert speed when distance is negative (robot should go forward) similar to backup recovery node.

## Description of documentation updates required from your changes

---

### CODE REVIEW

**Comment by SteveMacenski** [comment]
This is not an intuitive solution. I think you'd be better off checking both variables for signs and making sure both are negative.

**Comment by mdtoom** [comment]
I agree, and updated, but shouldn't then the backup recovery in nav2_recoveries also be changed because there a negative distance and a positive speed is required to go backward. This also not very intuitive.

**Comment by SteveMacenski** [comment]
I agree that's unintuitive for the API. Can you update the recovery behavior & the BT node to use only positive values as inputs and then set the negative signs required in the recovery behavior itself?

**Comment by mdtoom** [comment]
Interface changes documented in: https://github.com/ros-planning/navigation.ros.org/pull/60

File: nav2_behavior_tree/plugins/action/back_up_action.cpp [hunk 1]
```diff
@@ -31,10 +31,9 @@ BackUpAction::BackUpAction(
   double speed;
   getInput("backup_speed", speed);
 
-  // silently fix, vector direction determined by distance sign
-  if (speed < 0.0) {
-    speed *= -1.0;
-  }
+  // Silently ensure, that speed and distance are positive values
+  speed = std::fabs(speed);
```
Comment (SteveMacenski) [inline nav2_behavior_tree/plugins/action/back_up_action.cpp:35]: This is not necessary if we do i in the backup recovery itself. We can remove these 3 lines in favor for the 2 you added in back_up.cpp

**Pull request merged**